source/common/common/bit_array.h

```diff
@@ -46,8 +46,8 @@ class BitArray {
         // subtract one to fit in 32 bits.
         mask_(static_cast<uint32_t>((static_cast<uint64_t>(1) << width) - 1)),
         num_items_(num_items) {
-    RELEASE_ASSERT(width <= MaxBitWidth, "Using BitArray with invalid parameters.");
```

Comment:
This has me worried a bit. Supplying a value over 32 would lead to undefined behavior. Looking at the code, this class is used in maglev LB and would need more than 2^32 hosts in a locality to be a problem.

I think at the very least we'd need a TODO to move this close to the maglev LB, so it is not reused in some other place where the overflow is more likely. Perhaps it is worth having an issue with all of this questionable places tracked, so we can clean-up at some point.